src/components/breadcrumbs/crumb/crumb.component.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import tagComponent, {
+  TagProps,
+} from "../../../__internal__/utils/helpers/tags";
+import StyledCrumb from "./crumb.style";
+
+export interface CrumbProps extends TagProps {
+  /** The href string for Crumb Link */
+  href: string;
+  /** The to string for Crumb Link */
+  to?: string;
+  /** This sets the Crumb to current, does not render Link */
+  isCurrent?: boolean;
+  /** The onClick callback for the Crumb Link */
+  onClick?: (
+    ev:
+      | React.MouseEvent<HTMLAnchorElement>
+      | React.MouseEvent<HTMLButtonElement>
+      | React.KeyboardEvent<HTMLAnchorElement>
+      | React.KeyboardEvent<HTMLButtonElement>,
+    id: string,
+    to?: string
+  ) => null;
```

Comment:
```suggestion
  /** A callback function to call when the Crumb link is clicked */
  handleClick?: () => void;
```
I have a couple of points with regarding this prop:

Since this function is being utilised as a callback, personally I wouldn't call this function `onClick` since technically this isn't a browser event handler and could be confusing.

Since we are also surfacing `crumbId` and `to` as props, consumers should have access to these values already in scope so we don't have to reference these again within the callback itself.

On line 44 I noticed we are also using the `text` prop as an identifier if `crumbId` is not set. I think I understand why you have done it this way since I'm assuming we shouldn't have: 

1. Multiple `Breadcrumb` components on the same page (should be unique to one page)
2. Multiple `Crumb`s with the same text
    
If this is the case, I would be tempted to remove `crumbId` and instead make the `text` prop a required prop, since I assume it shouldn't be possible to have a crumb with no text :)

Reply:
You can run out of sensible synonyms quite quickly in that case :)

Reply:
It needs to be onClick so we can maintain the current approach for using routing solutions, that approach requires the event. Eventually we should investigate if we can adopt the approach used in the VerticalMenu for passing router links in Carbon's Link component but for now we should make sure our doc examples work